core/translate/select.rs

```diff
@@ -511,7 +524,22 @@ fn translate_tables_begin(program: &mut ProgramBuilder, select: &mut Select) ->
     let processed_where = process_where(program, select)?;
 
     for loop_info in &select.loops {
-        translate_table_open_loop(program, select, loop_info, &processed_where)?;
+        // if there is a left join and there is a condition on the join that is always false,
+        // every row in the outer table will be emitted with nulls for the right table
+        let early_terminate_label = if let Some(ljbk) = &loop_info.left_join_bookkeeping {
+            ljbk.check_match_flag_label
+        } else {
```

Comment:
I am not sure I understand this. If `left_join_bookkeeping` is `None`, then condition is always false?

Reply:
This `if/else` decides where to jump _if_ there is a condition on this loop that is always false.

If it's a condition on the LEFT JOIN (e.g. `x LEFT JOIN y ON false`), then it can't jump to e.g. `Halt`, but instead needs to emit nulls for the right table for every row in the left table.

If there is a condition in an INNER JOIN (or WHERE) that is always false, then the query will not produce any rows and it should jump to `Halt` (or e.g. `AggFinal` in case we have an aggregation expression like `count()` that should produce a `0` on empty input.

Maybe the above should be a code comment on this :)

Reply:
One more thing, just for explicitness: currently the existence of `Some(left_join_bookkeeping)` means that particular loop is a left join

Reply:
rewrote the code comment